src/binary-reader-ir.cc

```diff
@@ -1358,15 +1366,49 @@ Result BinaryReaderIR::EndDataSegmentInitExpr(Index index) {
   return EndInitExpr();
 }
 
+static Offset GetDataSegmentOffset(const DataSegment* data_segment) {
+  if (data_segment->offset.size() != 1) {
+    return kInvalidOffset;
+  }
+  const auto& expr = *data_segment->offset.begin();
+  if (expr.type() != ExprType::Const) {
+    return kInvalidOffset;
+  }
+  const Const& const_ = cast<ConstExpr>(&expr)->const_;
+  // clang-format off
```

Comment:
nit: maybe it is fine to clang-format here?

Reply:
done